scylla/src/transport/cluster.rs

```diff
@@ -481,6 +458,21 @@ impl ClusterData {
     pub fn replica_locator(&self) -> &ReplicaLocator {
         &self.locator
     }
+
+    /// Returns iterator of connections to all shards.
+    ///
+    /// By nonemptiness invariant of `self.known_peers`, the returned iterator is nonempty.
+    pub(crate) fn iter_working_connections(
```

Comment:
I have a slight concern about the naming choice in this piece of code. It's not immediately obvious to me why each element in the iterator that produces working connections needs to be a `Result` rather than an `Arc<Connection>`.

Reply:
Solved.